src/server/protocol.ts

```diff
@@ -672,6 +696,24 @@ export interface GetEditsForRefactorResponse extends Response {
     body?: RefactorEditInfo;
 }
 
+export interface GetEditsForMoveToFileRefactorRequest extends Request {
+    command: CommandTypes.GetEditsForMoveToFileRefactor;
+    arguments: GetEditsForMoveToFileRefactorRequestArgs;
+}
+
+export interface GetEditsForMoveToFileRefactorResponse extends Response {
+    body?: RefactorEditInfo; // TODO: maybe use a new type
+}
+
+export type GetEditsForMoveToFileRefactorRequestArgs = FileLocationOrRangeRequestArgs & {
+    /* The 'name' property from the refactoring that offered this action */
+    refactor: string;
+    /* The 'name' property from the refactoring action */
+    action: string;

```

Comment:
Are the `refactor` and `action ` properties needed? Or can we assume they are `Move to file` since we're making this call

Reply:
Yes, I believe the `refactor` and the `action` properties are needed to compute the associated code action. `getEditsForAction` in types.ts matches the `action` property to the correct refactor action.

Reply:
For this new API though, won't the action always be `Move to file`?

Reply:
I said today that my API questions didn’t affect the protocol, but I was kind of wrong; there is a parallel. My suggestion on the LS side was to remove the move-to-file-specific LS method and allow an extra argument to be passed to the existing `getEditsForRefactor` method. The exact same possibility exists here: instead of making a new command, we could reuse the existing `GetEditsForRefactor` command if we modify its request type to allow an argument containing the `targetFile` property. @mjbvz thoughts?

Reply:
I'm good with that. For the protocol, I'll just need the expected argument's type